src/html-to-ast.ts

```diff
@@ -0,0 +1,197 @@
+// LICENSE : MIT
+import type {
+    TxtBlockQuoteNode,
+    TxtBreakNode,
+    TxtCodeBlockNode,
+    TxtCodeNode,
+    TxtCommentNode,
+    TxtDeleteNode,
+    TxtDocumentNode,
+    TxtEmphasisNode,
+    TxtHeaderNode,
+    TxtHorizontalRuleNode,
+    TxtHtmlNode,
+    TxtImageNode,
+    TxtLinkNode,
+    TxtListItemNode,
+    TxtListNode,
+    TxtParagraphNode,
+    TxtParentNode,
+    TxtStrNode,
+    TxtStrongNode,
+    TxtTableCellNode,
+    TxtTableNode,
+    TxtTableRowNode
+} from "@textlint/ast-node-types";
+import { unified } from 'unified'
+import rehypeParse from 'rehype-parse'
+import traverse, { TraverseContext } from "traverse";
+import { StructuredSource } from "structured-source";
+import { nodeTypes, tagNameToType } from "./mapping.js";
+import { Element, RootContent } from "hast";
+
+/**
+ * Remove undocumented properties on TxtNode from node
+ */
+// function removeUnusedProperties(node: Element) {
+//     if (typeof node !== "object") {
+//         return;
+//     }
+//     ["position"].forEach(function (key) {
+//         if (node.hasOwnProperty(key)) {
+//             // @ts-expect-error: delete key
+//             delete node[key];
+//         }
+//     });
+// }
+
+function mapNodeType(node: RootContent, parent: TraverseContext | undefined) {
+    if (parent) {
+        const parentNode = parent.parent?.node;
+        if (parentNode?.tagName === "script" || parentNode?.tagName === "style") {
+            return "CodeBlock";
+        }
+    }
+    if (node.type === "element") {
+        // @ts-expect-error: tagName is string
+        const mappedType = tagNameToType[node.tagName];
+        if (mappedType) {
+            // p => Paragraph...
+            return mappedType;
+        } else {
+            // other case, return original tagName
+            return node.tagName;
+        }
+    } else if (node.type === "doctype") {
+        return "doctype";
```

Comment:
This will fix #19 

`html` to be `html`
Previously, `html` to be `Html`. This cause https://github.com/textlint-ja/textlint-rule-no-synonyms/issues/4